src/numeric/processes.jl

```diff
@@ -110,116 +129,59 @@ end
-
-        # Get transition probabilities
-
-        integration_weights = repmat(kron([w[i] for i in 1:n]...), 1, kron([n_integration[i] for i in 1:n]...))
-        integration_weights =collect(integration_weights[:, ii] for ii in 1:kron([n_integration[i] for i in 1:n]...))
-      else
-        yⁱʲ  = zeros(n_integration[n], n, n_states[n]);
-
-        for ii in 1:n_states[n]
-            [ yⁱʲ[ss, :, ii] = VAR_info.M + VAR_info.R*collect(yⁱ[ii])+collect(Eⁱʲ_c[ss]) for ss = 1:n_integration[n]]
-
-        end
-        integration_nodes = collect(yⁱʲ[:, :, ii] for ii in 1:n_states[n])
-
-        # Get transition probabilities
-
-        integration_weights = w[n]
-        integration_weights =[integration_weights]
-      end
-#  return nodes, integration_nodes, integration_weights
-    return DiscretizedProcess(nodes, integration_nodes, integration_weights)
+function discretize(var::VAR1, n_states::Array{Int64,1}, n_integration::Array{Int64,1}; n_std::Int64=2,)
+
+    R = var.R
+    M = var.M
+    Sigma = var.Sigma
+    S = QE.solve_discrete_lyapunov(R,Sigma)  # asymptotic variance
+    sig = diag(S)
+    min = var.M - n_std*sqrt(sig)
+    max = var.M + n_std*sqrt(sig)
+    grid = Dolo.CartesianGrid(min,max,n_states)
+    # discretize innovations
+    x,w = QE.qnwnorm(n_integration, zeros(size(var.Sigma,1)), var.Sigma)
+    integration_nodes = [ cat(1,[(M + R*(Dolo.node(grid, i)-M) + x[j,:])' for j=1:size(x,1)]...) for i in 1:Dolo.n_nodes(grid)]
+    integration_weights = [w for i in 1:Dolo.n_nodes(grid)]
+    return DiscretizedProcess(grid, integration_nodes, integration_weights)
 end
 
 
 
-function simulate_var(VAR_info::VAR1, T::Int64, N::Int64)
+function simulate(var::VAR1, N::Int64, T::Int64)
 
   """
   This function takes:
     - VAR_info -  an object of the type "VAR1.myvar"
     which contains 3 arrays: mean and autocorrelation matrices of the VAR(1) process
       and covariance matrix of the innovations;
-    - T - number of periods to simulate;
     - N - number of simulations.
+    - T - number of periods to simulate;
 
   The function returns:
-    - XN - simulated data, ordered as [N, T, n], where n is the number of variables;
-    - E - simulated innovations, ordered as [N, T, n], where n is the number of variables;
+    - XN - simulated data, ordered as [n, N, T], where n is the number of variables;
+    #- E - simulated innovations, ordered as [N, T, n], where n is the number of variables;
   """
-  n = size(VAR_info.M, 1);
-  srand(123) # Setting the seed
-  d = MvNormal(VAR_info.Sigma);
+  n = size(var.M, 1);
+  # srand(123) # Setting the seed
+  d = MvNormal(var.Sigma);
   XN=zeros(N, T, n);
   E=zeros(N, T, n);
   #Inicial conditions
   XN[:, 1, :]=rand(N, n);
   for jj = 1:1:N
         E[jj, :, :] = rand(d, T)';
         for ii =1:1:T-1
-        XN[jj, ii+1, :]=VAR_info.M+VAR_info.R*(XN[jj, ii, :]-VAR_info.M)+E[jj, ii, :];
+            XN[jj, ii+1, :]=var.M+var.R*(XN[jj, ii, :]-var.M)+E[jj, ii, :];
```

Comment:
I totally agree that we don't need to rush and change all the C-ordered algorithms in Dolo.jl, but I think here it is a no-brainer to change it to be `(n, T, N)` the whole time.